stage4.py

```diff
@@ -0,0 +1,48 @@
+from card import Card
+import random
+
+
+deck = list()
+cards_in_deck = 0
+is_card_identical = 0
+while cards_in_deck < 40:
+    pulled_card = Card()
+    for card in deck:
+        if card == pulled_card:
+            is_card_identical = 1
+            break
+    if is_card_identical == 1:
+        is_card_identical = 0
+        continue
+    else:
+        cards_in_deck += 1
```

Comment:
Storing two things that save the same information might lead to bugs because you must keep both of them all the time sync, for example, if something changes deck, he must change cards_in_deck, and if someone changes cards_in_deck he must change deck. 

A more safe approach would be to let anyone change the deck and in case you want to know how many cards you have inside the deck you can use 
`len(deck) `